HPXMLtoOpenStudio/tests/test_validation.rb

```diff
@@ -0,0 +1,618 @@
+      ['/HPXML/Building/BuildingDetails/MiscLoads/FuelLoad[FuelLoadType="lighting"]/SystemIdentifier'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/MiscLoads/FuelLoad[FuelLoadType="grill" or FuelLoadType="lighting" or FuelLoadType="fireplace"]/SystemIdentifier',
+      ['/HPXML/Building/BuildingDetails/MiscLoads/FuelLoad[FuelLoadType="fireplace"]/SystemIdentifier'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/MiscLoads/FuelLoad[FuelLoadType="grill" or FuelLoadType="lighting" or FuelLoadType="fireplace"]/SystemIdentifier',
+    }
+
+    expected_error_msgs_optional = {
+      ['/HPXML/SoftwareInfo/extension/SimulationControl', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/SoftwareInfo/extension/SimulationControl',
+      ['/HPXML/Building/BuildingDetails/BuildingSummary/Site', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/BuildingSummary/Site',
+      ['/HPXML/Building/BuildingDetails/BuildingSummary/BuildingOccupancy', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/BuildingSummary/BuildingOccupancy',
+      ['/HPXML/Building/BuildingDetails/ClimateandRiskZones/ClimateZoneIECC', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/ClimateandRiskZones/ClimateZoneIECC',
+      ['/HPXML/Building/BuildingDetails/Systems/HVAC/HVACControl', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Systems/HVAC/HVACControl',
+      ['/HPXML/Building/BuildingDetails/Systems/SolarThermal/SolarThermalSystem', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Systems/SolarThermal/SolarThermalSystem',
+      ['/HPXML/Building/BuildingDetails/Appliances/ClothesWasher', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/ClothesWasher',
+      ['/HPXML/Building/BuildingDetails/Appliances/ClothesDryer', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/ClothesDryer',
+      ['/HPXML/Building/BuildingDetails/Appliances/Dishwasher', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/Dishwasher',
+      ['/HPXML/Building/BuildingDetails/Appliances/Dehumidifier', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/Dehumidifier',
+      ['/HPXML/Building/BuildingDetails/Appliances/CookingRange', nil] => 'Expected 0 or 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/CookingRange',
+      ['/HPXML/Building/BuildingDetails/Appliances/Oven', nil] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/CookingRange/../Oven',
```

Comment:
Why are there some rows like this that include "Expected 1", which implies they are required, but they are located in the expected_error_msgs_optional hash instead of expected_error_msgs?

Reply:
Oops.  This looks wrong.  The effort to remove this huge hash is expected to eliminate this kind of errors.